packages-exp/auth-exp/src/core/auth/auth_event_manager.ts

```diff
@@ -18,43 +18,108 @@
 import {
   AuthEvent,
   AuthEventConsumer,
+  AuthEventType,
   EventManager
 } from '../../model/popup_redirect';
 import { AUTH_ERROR_FACTORY, AuthErrorCode } from '../errors';
 
 export class AuthEventManager implements EventManager {
   private readonly consumers: Set<AuthEventConsumer> = new Set();
+  private queuedRedirectEvent: AuthEvent | null = null;
+  private hasHandledPotentialRedirect = false;
 
   constructor(private readonly appName: string) {}
 
   registerConsumer(authEventConsumer: AuthEventConsumer): void {
     this.consumers.add(authEventConsumer);
+
+    if (
+      this.queuedRedirectEvent &&
+      this.isEventForConsumer(this.queuedRedirectEvent, authEventConsumer)
+    ) {
+      this.sendToConsumer(this.queuedRedirectEvent, authEventConsumer);
+      this.queuedRedirectEvent = null;
+    }
   }
 
   unregisterConsumer(authEventConsumer: AuthEventConsumer): void {
     this.consumers.delete(authEventConsumer);
   }
 
-  onEvent(event: AuthEvent): void {
+  onEvent(event: AuthEvent): boolean {
```

Comment:
this is ok... potentially worth refactoring eventually to handle redirect events in a completely separate path from popup events

Reply:
Agreed. Added a comment in popup_redirect.ts